navit/transform.c

```diff
@@ -570,7 +577,15 @@ int transform(struct transformation *t, enum projection required_projection, str
             clip_result=transform_z_clip_if_necessary(rotated_coord, zlimit, clip_result_old);
             clip_result_old=clip_result;
             if(clip_result.process_coord_again) {
-                i--;
+                loop_repeats++;
+                /* if we repeat an interation, we have to make sure that there is enough space in the result buffer to
+                   not overflow. */
+                if (i + loop_repeats < max_results) {
```

Comment:
You're checking against the wrong limit here. "i" isn't your concern, as "i" refers to your input buffer only.
If you think that both your input and output are 3 bytes long. Assume no points get skipped. the repeating the first point will be OK passing your check. But it will crash at the last point which is not repeated at all but still beyond array limits.

You should check "result_idx" against "max_result" instead, i think.
The you don't even need to count the repetitions in loop_repeats,

Line 610 in this listing including your changes is where you may want to look at. unfortunately this review tool doesn't allow me to annotate unchanged code. What I find a pity.

Reply:
Yeah, it was a bit overcomplicated. Just using `result_idx` is clearer to understand.